File: sdks/java/basys.sdk/src/main/java/org/eclipse/basyx/submodel/metamodel/map/qualifier/Description.java
package org.eclipse.basyx.submodel.metamodel.map.qualifier;

import java.util.HashMap;



/**
 * AAS description class
 * 
 * @author kuhn
 *
 */
public class Description extends HashMap<String, Object> {

		
	/**
	 * Version of serialized instances
	 */
	private static final long serialVersionUID = 1L;

	public static final String LANGUAGE = "language";
	public static final String TEXT = "text";
	

	/**
	 * Constructor
	 */
	public Description() {
		// Add qualifiers
		put(LANGUAGE, "");
		put(TEXT, "");
	}
	
	public Description(String lang,String text) {
		// Add qualifiers
		put(LANGUAGE, lang);
		put(TEXT, text);
	}

	public String getLanguage() {
		return (String) get(LANGUAGE);
	}

	public String getText() {
		return (String) get(TEXT);
	}
}

File: sdks/c++/basys.sdk.cc/src/aas/impl/metamodel/hashmap/aas/qualifier/Description.cpp
/*
 * Description.cpp
 *
 *      Author: wendel
 */

#include "Description.h"


namespace basyx {
namespace aas {
namespace qualifier {
namespace impl {

Description::Description(const std::string & language, const std::string & text) :
  language(language),
  text(text)
{}

Description::Description(basyx::objectMap_t & map)
{
  this->language = map.at(descriptionPaths::LANGUAGE).GetStringContent();
  this->text = map.at(descriptionPaths::TEXT).GetStringContent();
}

std::string Description::getLanguage() const
{
  return this->language;
}

std::string Description::getText() const
{
  return this->text;
}

}
}
}
}